src/EditorFeatures/Text/Extensions.TextBufferMapper.cs

```diff
@@ -16,48 +14,32 @@ public static partial class Extensions
         /// </summary>
         private static class TextBufferMapper
         {
-            private static readonly ConditionalWeakTable<ITextSnapshot, ITextSnapshot> s_editorToRoslynSnapshotMap = new ConditionalWeakTable<ITextSnapshot, ITextSnapshot>();
-            private static readonly ConditionalWeakTable<ITextSnapshot, WeakReference<ITextSnapshot>> s_roslynToEditorSnapshotMap = new ConditionalWeakTable<ITextSnapshot, WeakReference<ITextSnapshot>>();
-            private static readonly ConditionalWeakTable<ITextSnapshot, ITextSnapshot>.CreateValueCallback s_createSnapshotCallback = CreateSnapshot;
+            private static readonly ConditionalWeakTable<ITextImage, WeakReference<ITextSnapshot>> s_roslynToEditorSnapshotMap = new ConditionalWeakTable<ITextImage, WeakReference<ITextSnapshot>>();
 
-            public static ITextSnapshot ToRoslyn(ITextSnapshot editorSnapshot)
+            public static ITextImage ToReverseMappedTextImage(ITextSnapshot editorSnapshot)
```

Comment:
Does the editor allow multiple snapshots to have the same ITextImage? Or does each snapshot have a unique instance that might actually hold the same underlying data?

My concern is previously if you went from a real ITextSnapshot -> Roslyn -> ITextSnapshot, I believe other than GC you should get the original one back. That invariant might be broken now.